src/parser/shared/modules/RegenResourceCapTracker.js

```diff
@@ -195,6 +195,16 @@ class RegenResourceCapTracker extends Analyzer {
     return event.resourceChange;
   }
 
+  /**
+   * IMPLEMENTME
+   * Some specs have modifiers to the base regen rate that apply before haste, which should
+   * be implemented here.
+   * @returns {number} Base regen rate of the resource.
+   */
+  getBaseRegenRate(){
+    return this.constructor.baseRegenRate;
```

Comment:
I don't like having 2 ways to do something, so I would have just this method that everyone must override, but better ask maintainers what the policies are =)